compile errors Arch

Hello,

interested in the project and have been trying to get a good install but haven't had much luck.

Currently on the ninja -C build 


✔≻ ninja -C build
ninja: Entering directory `build'
[262/365] Compiling C object subprojects/wlroots-0.12.0/examples/dmabuf-capture.p/dmabuf-capture.c.o
FAILED: subprojects/wlroots-0.12.0/examples/dmabuf-capture.p/dmabuf-capture.c.o 
cc -Isubprojects/wlroots-0.12.0/examples/dmabuf-capture.p -Isubprojects/wlroots-0.12.0/examples -I../subprojects/wlroots-0.12.0/examples -Isubprojects/wlroots-0.12.0/protocol -I/usr/include/libdrm -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -g -DWLR_USE_UNSTABLE -Wundef -Wlogical-op -Wmissing-include-dirs -Wold-style-definition -Wpointer-arith -Winit-self -Wstrict-prototypes -Wimplicit-fallthrough=2 -Wendif-labels -Wstrict-aliasing=2 -Woverflow -Wmissing-prototypes -Wno-missing-braces -Wno-missing-field-initializers -Wno-unused-parameter -fmacro-prefix-map=../subprojects/wlroots-0.12.0/= -DLIBINPUT_MAJOR=1 -DLIBINPUT_MINOR=17 -DLIBINPUT_PATCH=2 '-DICONDIR="/usr/local/share/icons"' -pthread -MD -MQ subprojects/wlroots-0.12.0/examples/dmabuf-capture.p/dmabuf-capture.c.o -MF subprojects/wlroots-0.12.0/examples/dmabuf-capture.p/dmabuf-capture.c.o.d -o subprojects/wlroots-0.12.0/examples/dmabuf-capture.p/dmabuf-capture.c.o -c ../subprojects/wlroots-0.12.0/examples/dmabuf-capture.c
../subprojects/wlroots-0.12.0/examples/dmabuf-capture.c: In function ‘vid_encode_thread’:
../subprojects/wlroots-0.12.0/examples/dmabuf-capture.c:494:4: error: ‘av_init_packet’ is deprecated [-Werror=deprecated-declarations]
  494 |    av_init_packet(&pkt);
      |    ^~~~~~~~~~~~~~
In file included from /usr/include/libavcodec/bsf.h:30,
                 from /usr/include/libavcodec/avcodec.h:44,
                 from /usr/include/libavformat/avformat.h:312,
                 from ../subprojects/wlroots-0.12.0/examples/dmabuf-capture.c:2:
/usr/include/libavcodec/packet.h:488:6: note: declared here
  488 | void av_init_packet(AVPacket *pkt);
      |      ^~~~~~~~~~~~~~
cc1: all warnings being treated as errors
[267/365] Linking target subprojects/wlroots-0.12.0/libwlroots.so.7
ninja: build stopped: subcommand failed.


Now before this started coming up I was able to compile and started a config but the executable now returns:
vivarium: error while loading shared libraries: libwlroots.so.7: cannot open shared object file: No such file or directory



I just cloned the sourced and tried to build, prior directory I just removed prior.

I know AUR is still not functioning due to wlroots, any suggestions?

Thanks....

hitime

--- a/meson.build
+++ b/meson.build
@@ -35,7 +35,8 @@ endif
 cc = meson.get_compiler('c')
 
 # For wlroots, use the system version if the correct version is available
-wlroots_dep = dependency('wlroots', version : '0.12.0', fallback : ['wlroots', 'wlroots'])
+wlroots_dep = dependency('wlroots', version : '0.12.0', fallback : ['wlroots', 'wlroots'],
+                         default_options : ['examples=false'])
 
 tomlc99_dep = dependency('toml', fallback : ['tomlc99', 'tomlc99_static_dep'])
 

